src/pages/index.js

```diff
@@ -64,68 +77,177 @@ const addCardFormValidator = new FormValidator(validationSettings, addCardForm);
 
 addCardFormValidator.enableValidation();
 
-const addNewCardPopup = new PopupWithForm("#add-card-modal", handleCardSubmit);
+const editAvatarFormValidator = new FormValidator(
+  validationSettings,
+  editAvatarForm
+);
+
+editAvatarFormValidator.enableValidation();
+
+const addNewCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleCardSubmit,
+  "Create"
+);
 addNewCardPopup.setEventListners();
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  handleProfileEditSubmit,
+  "Save"
 );
 editProfilePopup.setEventListners();
 
 const imagePreviewPopup = new PopupWithImage("#preview-image-modal");
 imagePreviewPopup.setEventListners();
 
+const confirmDelete = new PopupWithConfirm({
+  popupSelector: "#confirm-delete-modal",
+  // handler shoould be here
+  handler: (cardId) => {},
+});
+
+const avatarModal = new PopupWithForm(
+  "#avatar-modal",
+  handleUpdateAvatar,
+  "Save"
+);
+avatarModal.setEventListners();
+
+confirmDelete.setEventListners();
+
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
   userDescription: ".profile__description",
+  userAvatar: ".profile__image",
 });
 
 function handleImageClick(data) {
   imagePreviewPopup.open(data);
 }
 
+function handleUpdateAvatar({ link }) {
+  avatarModal.renderLoading(true);
+  api
+    .changeProfilePic({ avatar: link })
+    .then((data) => {
+      avatarModal.close();
+      userInfo.setUserInfo(data.name, data.description, data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarModal.renderLoading(false);
+    });
+}
+
 function createCard(data) {
-  return new Card(data, cardSelector, handleImageClick).getView();
+  const card = new Card(
+    data,
+    cardSelector,
+    handleImageClick,
+    (card) => {
+      confirmDelete.open();
+      confirmDelete.setSubmitAction(() => {
+        return api
+          .deleteCard(card.getId())
+          .then(() => {
+            card.handleDeleteCard();
+            confirmDelete.close();
+          })
+          .catch((err) => {
+            alert("No Card Available", err);
+          });
+      });
+    },
+    (card) => {
+      api
+        .changeLikeCardStatus(card.getId(), !card.isLiked())
+        .then(() => {
+          card.handleLikeIcon();
+        })
+        .catch((err) => {});
```

Comment:
`        .catch((err) => {});
`

Please, handle the error in any way

At least this way:

```js
.catch(console.error);

```

Reply:
Please, add correct error handling